WebApiCoreSeed.WebApi.IntegrationTests/Users/UserIntegrationTest.cs

```diff
@@ -0,0 +1,285 @@
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using WebApiCoreSeed.Data.EF;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.WebApi.IntegrationTests.Fake;
+using WebApiCoreSeed.WebApi.IntegrationTests.Users.TestData;
+using WebApiCoreSeed.WebApi.Models;
+using Xunit;
+
+namespace WebApiCoreSeed.WebApi.IntegrationTests.Users
+{
+    public class UserIntegrationTest
+    {
+        private const string ResouceUri = "api/users/";
+
+        private readonly TestServer _server;
+        private readonly HttpClient _client;
+
+        private readonly DbContextOptions<WebApiCoreSeedContext> _dbContextOptions;
+
+        public UserIntegrationTest()
+        {
+            FileFaker.Fake();
+
+            var builder = new WebHostBuilder()
+                .UseEnvironment("development")
+                .UseStartup<Startup>();
+
+            _server = new TestServer(builder);
+            _client = _server.CreateClient();
+            _dbContextOptions = _server.Host.Services
+                .GetService(typeof(DbContextOptions<WebApiCoreSeedContext>)) as DbContextOptions<WebApiCoreSeedContext>;
```

Comment:
Let's create a Fixture that contains all this initialization code. Otherwise we will have this code duplicated for all new classes tests.

Reply:
Something like:

```csharp
public class ApiTestFixture
{
	private readonly TestServer _server;

	public ApiTestFixture()
	{
		var builder = new WebHostBuilder()
			.UseEnvironment("development")
			.UseStartup<Startup>();
			
		_server = new TestServer(builder);
		
		HttpClient = _server.CreateClient();
	}
	
	public HttpClient HttpClient { get; }
}

public class UserControllerTests : IClassFixture<ApiTestFixture>
{
	private readonly HttpClient _httpClient;
	
	public AuthControllerTests(ApiTestFixture fixture)
	{
		_httpClient = fixture.HttpClient;
	}
}
```